src/uucore/src/lib/features/entries.rs

```diff
@@ -95,15 +95,18 @@ pub fn get_groups() -> IOResult<Vec<gid_t>> {
 /// history of a process and its parents could affect the details of
 /// the result.)
 pub fn get_groups_gnu(arg_id: Option<u32>) -> IOResult<Vec<gid_t>> {
-    let mut groups = get_groups()?;
+    let groups = get_groups()?;
     let egid = arg_id.unwrap_or_else(crate::features::process::getegid);
-    if !groups.is_empty() && *groups.first().unwrap() == egid {
-        return Ok(groups);
-    } else if let Some(index) = groups.iter().position(|&x| x == egid) {
-        groups.remove(index);
+    Ok(sort_groups(groups, egid))
+}
+
+fn sort_groups(mut groups: Vec<gid_t>, egid: gid_t) -> Vec<gid_t> {
+    if let Some(index) = groups.iter().position(|&x| x == egid) {
+        groups[..index + 1].rotate_right(1);
```

Comment:
You can write this with an inclusive range:
```suggestion
        groups[..=index].rotate_right(1);
```